core/src/main/java/org/elasticsearch/index/query/RegexpQueryBuilder.java

```diff
@@ -19,38 +19,136 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.RegexpQuery;
 import org.apache.lucene.util.automaton.Operations;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.BytesRefs;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
  * A Query that does fuzzy matching for a specific value.
  */
-public class RegexpQueryBuilder extends MultiTermQueryBuilder {
+public class RegexpQueryBuilder extends MultiTermQueryBuilder<RegexpQueryBuilder> {
 
     public static final String NAME = "regexp";
-    private final String name;
-    private final String regexp;
 
-    private int flags = -1;
+    public static final int DEFAULT_FLAGS_VALUE = RegexpFlag.ALL.value();
+
+    public static final int DEFAULT_MAX_DETERMINIZED_STATES = Operations.DEFAULT_MAX_DETERMINIZED_STATES;
+
+    private final String fieldName;
+    
+    private final Object value;
```

Comment:
The documentation mentions that default regex flag is ALL which isn't ordinal value of -1.